core/modules/memman/MemFile.h

```diff
@@ -0,0 +1,138 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2016 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#ifndef LSST_QSERV_MEMMAN_MEMFILE_H
+#define LSST_QSERV_MEMMAN_MEMFILE_H
+
+// System headers
+#include <mutex>
+#include <string>
+#include <unistd.h>
+
+// Qserv headers
+#include "memman/Memory.h"
+
+// System headers
+#include <atomic>
+#include <string>
+
+namespace lsst {
+namespace qserv {
+namespace memman {
+
+//-----------------------------------------------------------------------------
+//! @brief Description of a memory based file.
+//-----------------------------------------------------------------------------
+
+class MemFile {
+public:
+
+    //-----------------------------------------------------------------------------
+    //! @brief Get file path for this table.
+    //!
+    //! @return The filename.
+    //-----------------------------------------------------------------------------
+
+    std::string filePath() {return _fPath;}
+
+    //-----------------------------------------------------------------------------
+    //! @brief Lock database file in memory.
+    //!
+    //! @param  maxBytes- maximum file size       for locking the table file.
+    //!                   A value of zero skips the check.
+    //! @param  minRefs - minimum reference count for locking the table file.
+    //!
+    //! @return MLResult  When bLocked > 0 this number of bytes locked.
+    //!                   When bLocked = 0 no bytes were locked and retc holds
+    //!                   the reason as follows:
+    //! @return =0        retc = 0 the file did not meet minRefs restriction;
+    //!                   otherwise, it holds the errno failure code.
+    //-----------------------------------------------------------------------------
+
+    struct MLResult {size_t bLocked;
+                     int    retc;
+                     MLResult() {}
+                     MLResult(size_t lksz, int rc) : bLocked(lksz), retc(rc) {}
+                    };
```

Comment:
size_t should be on a new line indented 4 spaces. other entries should lineup with size_t, end bracket should line up with struct.